hazelcast/src/main/java/com/hazelcast/internal/management/operation/ChangeWanStateOperation.java

```diff
@@ -21,19 +21,20 @@
 import com.hazelcast.wan.WanReplicationService;
 
 /**
- * Enables/Disable publishing events to target cluster in WAN Replication {@link com.hazelcast.wan.WanReplicationService}
- * on a node. This operation does not block adding new events to event queue.
+ * Stop, pause or resume WAN replication for the given {@code wanReplicationName} and {@code targetGroupName}.
  */
 public class ChangeWanStateOperation extends AbstractLocalOperation {
 
     private String schemeName;
     private String publisherName;
     private boolean start;
+    private boolean stop;
```

Comment:
Do we break some compatibility guarantee if we substitute the `start` field with a `WanPublisherState state`? Just wondering, since with 2 booleans, there might be some ambiguous states like `start == true && stop == true`.

Reply:
I don't think we'd break compatibility as this is a local operation. We do need to have as two booleans on `ChangeWanStateRequest` though, because of compatibility. We could have the state in `ChangeWanStateOperation` but I thought keeping them in sync is better.

For ambiguity, I agree with you that this should be a state and MC should request one of the states to be the new one. The logical UI element for this on MC would be a dropdown where you select a new state. I suggested having this as a dropdown where you can select a state but @dbrimley wants to have it as a switch for start/stop, and keep the current pause/start logic. So at any time, MC will either tell start, stop or pause. See the screenshot below to better understand what I mean:

![screenshot from 2018-06-11 10-41-51](https://user-images.githubusercontent.com/719018/41218527-18cc9f92-6d64-11e8-8f28-cc2404fc2d9c.png)